(#1102) toString() implementation for Immutable iterator

scristalli opened the pull request:
This is for #1102 

As discussed in the issue, `toString()` implementation for `Immutable` should be delegated to its inner iterator.

0crat commented:
@scristalli[/z](https://www.0crat.com/u/scristalli) this pull request is too small, just 4 lines changed (less than 10), there will be no formal code review, see [§53](http://www.zerocracy.com/policy.html#53) and [§28](http://www.zerocracy.com/policy.html#28); in the future, try to make sure your pull requests are not too small; @llorllale[/z](https://www.0crat.com/u/llorllale) please review this and merge or reject

<!-- https://www.0crat.com/footprint/C63314D6Z/196d5768-5b90-4552-b00a-e614fb5555df, version: 0.47.5, hash: ${buildNumber} -->

scristalli commented:
@llorllale is it possible to have this merged, or get review started since with the second commit the edit is more than 10 lines? Thank you.

llorllale commented on src/test/java/org/cactoos/iterator/ImmutableTest.java:
```diff
@@ -66,4 +67,26 @@ public void decoratesHasNext() {
+                return string;
+            }
+        };
+        final Iterator<Object> immutable = new Immutable<>(iterator);
```
@scristalli use `Assertion`

scristalli replied:
@llorllale done

llorllale commented:
@rultor merge

rultor commented:
> @rultor merge

@llorllale OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/19251-503158320)

rultor commented:
> @rultor merge

@llorllale @scristalli Oops, I failed. You can see the full log [here](http://www.rultor.com/t/19251-503158320) (spent 7min)

```
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 05:03 min
[INFO] Finished at: 2019-06-18T14:30:49+00:00
[INFO] Final Memory: 33M/441M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:testCompile (default-testCompile) on project cactoos: Compilation failure
[ERROR] /home/r/repo/src/test/java/org/cactoos/iterator/ImmutableTest.java:[92,22] incompatible types: cannot infer type arguments for org.llorllale.cactoos.matchers.Assertion<>
[ERROR] reason: cannot infer type-variable(s) T
[ERROR] (argument mismatch; bad return type in lambda expression
[ERROR] org.cactoos.text.TextOf cannot be converted to java.lang.String)
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:testCompile (default-testCompile) on project cactoos: Compilation failure
/home/r/repo/src/test/java/org/cactoos/iterator/ImmutableTest.java:[92,22] incompatible types: cannot infer type arguments for org.llorllale.cactoos.matchers.Assertion<>
    reason: cannot infer type-variable(s) T
      (argument mismatch; bad return type in lambda expression
        org.cactoos.text.TextOf cannot be converted to java.lang.String)

	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.compiler.CompilationFailureException: Compilation failure
/home/r/repo/src/test/java/org/cactoos/iterator/ImmutableTest.java:[92,22] incompatible types: cannot infer type arguments for org.llorllale.cactoos.matchers.Assertion<>
    reason: cannot infer type-variable(s) T
      (argument mismatch; bad return type in lambda expression
        org.cactoos.text.TextOf cannot be converted to java.lang.String)

	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1215)
	at org.apache.maven.plugin.compiler.TestCompilerMojo.execute(TestCompilerMojo.java:181)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 20 more
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
container 599f96ec723e9051d8240f01ed64c054dbc345d58f5cd43296629485d24f31fc is dead
Tue Jun 18 16:31:38 CEST 2019
```

scristalli commented:
@llorllale the merge didn't work due to some changes that have been pushed to master in the meanwhile. I rebased my branch and changed the `Assertion`, should work now.

llorllale commented:
@rultor merge

rultor commented:
> @rultor merge

@llorllale OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/19251-503245819)

rultor commented:
> @rultor merge

@llorllale Done! FYI, the full log is [here](http://www.rultor.com/t/19251-503245819) (took me 15min)

0crat commented:
Job `gh:yegor256/cactoos#1138` is not assigned, can't get performer

<!-- https://www.0crat.com/footprint/C63314D6Z/1acfbeaf-13cc-4b9d-8e7e-a9dd24f9aac5, version: 0.47.6, hash: ${buildNumber} -->

0crat commented:
This job is not in scope

<!-- https://www.0crat.com/footprint/C63314D6Z/aacd811a-163b-4913-9577-c47206142f84, version: 0.47.6, hash: ${buildNumber} -->
